docs/topics/control-flow.md

```diff
@@ -177,6 +177,82 @@ fun Request.getBody() =
 
 The scope of variable introduced in *when* subject is restricted to the body of this *when*.
 
+### Guard conditions in when expressions
+
+> Guard conditions are an [experimental feature](components-stability.md#stability-levels-explained) that may be changed at any time.
+> We would appreciate your feedback in [YouTrack](https://youtrack.jetbrains.com/issue/KT-71140/Guard-conditions-in-when-expressions-feedback).
+>
+{type="warning"}
+
+Starting from Kotlin 2.1, you can use guard conditions in `when` expressions or statements with a subject.
+
+Guard conditions allow you to include more than one condition to the branches of a `when` expression, making complex control flow more explicit and concise.
+
+To include a guard condition in a branch, place the guard condition after the primary condition, separated by `if`:
+
+```kotlin
+sealed interface Animal {
+    data class Cat(val mouseHunter: Boolean) : Animal
+    data class Dog(val breed: String) : Animal
+}
+
+fun feedAnimal(animal: Animal) {
+    when (animal) {
+        // Branch with only primary condition. Returns `feedDog()` when `Animal` is `Dog`
+        is Animal.Dog -> feedDog()
+        // Branch with both primary and guard conditions. Returns `feedCat()` when `Animal` is `Cat` and is not `mouseHunter`
+        is Animal.Cat if !animal.mouseHunter -> feedCat()
+        // Returns "Unknown animal" if none of the above conditions match
+        else -> println("Unknown animal")
+    }
+}
+```
+
+In a single `when` expression, you can combine branches with and without guard conditions. 
+The code in a branch with a guard condition runs only if both the primary condition and the guard condition evaluate to true.
+If the primary condition does not match, the guard condition is not evaluated. 
+
+If you use guard conditions in `when` statements without an `else` branch, if none of the conditions matches, none of the branches is executed. 
```

Comment:
I'd like to avoid "if" doubling. 
Can we say: If you use guard conditions in `when` statements without an `else` branch, and none of the conditions match, no branches are executed.